server/src/test/java/keywhiz/service/daos/SecretSeriesDAOTest.java

```diff
@@ -168,6 +168,73 @@ public void setCurrentVersion_failsWithIncorrectSecretContent() {
     assertThat(secretSeriesDAO.getSecretSeriesById(id)).isEmpty();
   }
 
+  @Test public void renameSecretSeriesById() {
+    long now = OffsetDateTime.now().toEpochSecond();
+    String oldName = "toBeRenamed_renameSecretSeriesById";
+    long id = secretSeriesDAO.createSecretSeries(oldName,
+        "creator", "", null, null, now);
+    long contentId = secretContentDAO.createSecretContent(id, "blah",
+        "checksum", "creator", null, 0, now);
+    secretSeriesDAO.setCurrentVersion(id, contentId, "creator", now);
+    assertThat(secretSeriesDAO.getSecretSeriesByName(oldName).get().currentVersion()).isPresent();
+
+    String newName = "newName";
+    secretSeriesDAO.renameSecretSeriesById(id, newName);
+    assertThat(secretSeriesDAO.getSecretSeriesByName(newName).get().currentVersion()).isPresent();
+  }
+
+  @Test public void updateSecretSeriesContentById() {
+    long now = OffsetDateTime.now().toEpochSecond();
+    long id = secretSeriesDAO.createSecretSeries("toBeUpdated_updateSecretSeriesContentById",
+        "creator", "", null, null, now);
+    long oldContentId = secretContentDAO.createSecretContent(id, "blah",
+        "checksum", "creator", null, 0, now);
+    secretSeriesDAO.setCurrentVersion(id, oldContentId, "creator", now);
+    assertThat(secretSeriesDAO.getSecretSeriesById(id).get().currentVersion().get())
+        .isEqualTo(oldContentId);
+
+    long newContentId = secretContentDAO.createSecretContent(id, "newblah",
+        "checksum", "creator", null, 0, now);
+
+    secretSeriesDAO.updateSecretSeriesContentById(id, newContentId);
+    assertThat(secretSeriesDAO.getSecretSeriesById(id).get().currentVersion().get())
+        .isEqualTo(newContentId);
+  }
+
+  @Test public void getSecretSeriesByDeletedName() {
+    long now = OffsetDateTime.now().toEpochSecond();
+    long id = secretSeriesDAO.createSecretSeries("toBeFound_getSecretSeriesByDeletedName",
+        "creator", "", null, null, now);
+    long oldContentId = secretContentDAO.createSecretContent(id, "blah",
+        "checksum", "creator", null, 0, now);
+    secretSeriesDAO.setCurrentVersion(id, oldContentId, "creator", now);
+    secretSeriesDAO.deleteSecretSeriesById(id);
+
+    List<SecretSeries> deletedSecretSeries =
+        secretSeriesDAO.getSecretSeriesByDeletedName("toBeFound_getSecretSeriesByDeletedName");
+    assertThat(deletedSecretSeries.size()).isEqualTo(1);
+    assertThat(deletedSecretSeries.get(0).name()).contains("toBeFound_getSecretSeriesByDeletedName");
+    assertThat(deletedSecretSeries.get(0).name()).isNotEqualTo("toBeFound_getSecretSeriesByDeletedName");
+    assertThat(deletedSecretSeries.get(0).id()).isEqualTo(id);
+  }
+
+  @Test public void getDeletedSecretSeriesById() {
+    long now = OffsetDateTime.now().toEpochSecond();
+    long id = secretSeriesDAO.createSecretSeries("toBeFound_getSecretSeriesByDeletedName",
```

Comment:
nit: the secret should probably be called `toBeFound_getSecretSeriesByDeletedId` so that it has a unique name compared to the other new test. Unfortunately, the table is not reset between tests, so duplicate secret names can cause unexpected behavior.